pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -116,10 +119,11 @@ func New(cidr *net.IPNet, allocatorFactory allocator.AllocatorWithOffsetFactory)
 	max--
 
 	r := Range{
-		net:    cidr,
-		base:   base,
-		max:    maximum(0, int(max)),
-		family: family,
+		net:     cidr,
+		base:    base,
+		max:     maximum(0, int(max)),
+		family:  family,
+		metrics: &emptyMetricsRecorder{}, // disabled by default
```

Comment:
then, should we `registerMetrics()` in L96, or only when we enable the metrics?

Reply:
Thank you. I moved it to in `EnableMetrics()`

https://github.com/kubernetes/kubernetes/compare/f2f925a67f91f496e3875b8a8131f7eb243e352e..187af7781a4d6a308ea7f73dd22d435575873396#diff-3f7680212eb37166b66a0f052153bb765c0d4b141af958d5b92543dda628d35fR373